### ISSUE TITLE
Self allocation reporting issue

### ISSUE THREAD

**Issue by Bojun-Seo** [honggyukim/heaptrace#15]
Way to reproduce
Compile the following code
just call `g++ a.cpp` will generate `a.out`
```
#include <iostream>
#include <unistd.h>
#include <malloc.h>

int period = 5;

void printErrorAndExit() {
  std::cerr << "Failed to allocate memory\n";
  exit(-1);
}

int* test_malloc() {
  int* tmp = (int*)malloc(sizeof(int));
  if (nullptr == tmp) {
    printErrorAndExit();
  }
  *tmp = 41;
  return tmp;
}

int main(int argc, char* argv[]) {
  int* val = nullptr;
  while (true) {
    val = test_malloc();
    *val = 33;
    sleep(period);
  }
  return 0;
}
```

```
$ ./heaptrace a.out
[heaptrace] initialized for /proc/2220/maps (a.out)
=================================================================
[heaptrace] dump allocation sorted by 'size' for /proc/2220/maps (a.out)
=== backtrace #1 === [count/peak: 1/1] [size/peak: 32 bytes/32 bytes] [age: 111.305 us]
 0 [0x7f51b508d47f] operator new(unsigned long) +0x1f
 1 [0x7f51b5093bc3] void std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11... +0x70
 2 [0x7f51b50938fe] ./libheaptrace.so (+0x7f51b50938fe)
 3 [0x7f51b5090487] ./libheaptrace.so (+0x7f51b5090487)
 4 [0x7f51b4c6f520] /lib/x86_64-linux-gnu/libc.so.6 (+0x7f51b4c6f520)
 5 [0x7f51b4d127fa] clock_nanosleep +0x16
 6 [0x7f51b4d176e7] __nanosleep +0x5
 7 [0x7f51b4d1761e] sleep +0xf

=== backtrace #2 === [count/peak: 7/7] [size/peak: 28 bytes/28 bytes] [age: 31.304 secs]
 0 [0x7f51b508d5ef] malloc +0x1f
 1 [0x559e9021722a] samples/a.out (+0x559e9021722a)
 2 [0x559e9021726a] samples/a.out (+0x559e9021726a)
 3 [0x7f51b4c56d90] /lib/x86_64-linux-gnu/libc.so.6 (+0x7f51b4c56d90)
 4 [0x7f51b4c56e40] __libc_start_main +0x20
 5 [0x559e90217125] samples/a.out (+0x559e90217125)

[heaptrace] heap traced num of backtrace : 2
[heaptrace] heap traced allocation size  : 60 bytes
[heaptrace] allocator info (virtual)     : 135.168 KB
[heaptrace] allocator info (resident)    : 80.496 KB
[heaptrace] statm info (VSS/RSS/shared)  : 6.246 MB / 1.966 MB / 1.777 MB
=================================================================
```

The report shows a heaptrace self allocated memory usage.

**Comment by honggyukim** [comment]
I've simplified your example as follows.
```c
$ cat alloc.c 
#include <stdio.h>
#include <stdlib.h>

#define KB (1024UL)
#define MB (KB * (1024UL))
#define GB (MB * (1024UL))

int main()
{
  char* p;

  while (1) {
    p = (char*)malloc(1 * GB);
    if (p == NULL) {
      fprintf(stderr, "Failed to allocate memory\n");
      return -1;
    }
    *p = 'X';
  }
  return 0;
}
```
It shows the same allocation failure message in the end.
```
$ gcc alloc.c

$ ./a.out 
Failed to allocate memory

$ heaptrace ./a.out 
[heaptrace] initialized for /proc/6807/maps (a.out)
Failed to allocate memory
[heaptrace]   finalized for /proc/6807/maps (a.out)
=================================================================
[heaptrace] dump allocation sorted by 'size' for /proc/6807/maps (a.out)
=== backtrace #1 === [count/peak: 131062/131062] [size/peak: 140726750.937 MB/140726750.937 MB] [age: 842.613 ms]
 0 [0x7f03b2c925ef] malloc +0x1f
 1 [0x55c6ddaa517f] ./a.out (+0x55c6ddaa517f)
 2 [0x7f03b2a29d90] /lib/x86_64-linux-gnu/libc.so.6 (+0x7f03b2a29d90)
 3 [0x7f03b2a29e40] __libc_start_main +0x20
 4 [0x55c6ddaa50a5] ./a.out (+0x55c6ddaa50a5)

=== backtrace #2 === [count/peak: 1/1] [size/peak: 32 bytes/32 bytes] [age: 18.226 us]
 0 [0x7f03b2c9247f] operator new(unsigned long) +0x1f
 1 [0x7f03b2c98bc3] void std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11... +0x70
 2 [0x7f03b2c988fe] ./libheaptrace.so (+0x7f03b2c988fe)
 3 [0x7f03b2c95487] ./libheaptrace.so (+0x7f03b2c95487)
 4 [0x7f03b2c91f2b] ./libheaptrace.so (+0x7f03b2c91f2b)
 5 [0x7f03b2ca524e] /lib64/ld-linux-x86-64.so.2 (+0x7f03b2ca524e)
 6 [0x7f03b2a45495] /lib/x86_64-linux-gnu/libc.so.6 (+0x7f03b2a45495)
 7 [0x7f03b2a45610] on_exit +0

[heaptrace] heap traced num of backtrace : 2
[heaptrace] heap traced allocation size  : 140726750.937 MB
[heaptrace] allocator info (virtual)     : 18446744072107.16 MB
[heaptrace] allocator info (resident)    : 18446744071582.68 MB
[heaptrace] statm info (VSS/RSS/shared)  : 18446744073709.547 MB / 18446744072678.440 MB / 134.213 MB
```
It looks the last statm info is incorrect. I will take a look later when I have more time.

**Comment by honggyukim** [comment]
It's fixed by d95b8c4.
